### DISCUSSION TITLE
Retrieving Custom FrontMatter?

### DISCUSSION THREAD

**Discussion by justinyoo** [statiqdev/Discussions#35] [Q&A] [answered]
Hi,

If I want to add a custom frontmatter key/value pair, like `CanonicalUrl` or `HeroImageUrl`, which object should I look for them? Thought it would be `WebKeys` or `BlogKeys`, but it wasn't.

The document, https://statiq.dev/web/content-and-data/metadata/front-matter, doesn't say how to access those values either.

**Comment by daveaglick** [comment, marked as answer]
Great question, I'll add some additional detail to the docs! The way front matter works is that it gets stored in the document as a key/value pair. The document is very much like a `Dictionary<string,object>` with some extra super-powers like handy type conversion. To put it another way, you use the name of the front matter as a string-based key anywhere you have an `IDocument`.

In a Razor page that might look like `Model["GetImageUrl"]` (since the `Model` property is set to the current document in Razor, though you could also use `Document` instead). We generally don't access document metadata using the raw indexer though, instead it's often easier to access via one of the special `Get...` type conversion extensions like `Model.GetString("GetImageUrl")`. There are lots of ways to access document metadata (I.e. front matter), so let me know if you're looking for something other than Razor.

**Reply by justinyoo** [reply]
Oh, there we go! I got what I want, with `@Document.GetString("CanonicalUrl")` in the Razor page. As I'm exploring Statiq, I think it's OK for now. Thanks for your great work!
